k2/csrc/self_alignment.h

```diff
@@ -0,0 +1,41 @@
+/**
+ * @copyright
+ * Copyright      2022  Xiaomi Corporation (authors: Liyong Guo)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef K2_CSRC_SELF_ALIGNMENT_H_
+#define K2_CSRC_SELF_ALIGNMENT_H_
+
+#include <torch/extension.h>
+
+#include <vector>
+
+#include "k2/python/csrc/torch.h"
+
+namespace k2 {
+
+FsaVec SelfAlignment(
```

Comment:
I think we would need more documentation here.  The names here seem to relate to the RNN-T code, and in addition to explaining what the various variables mean you should point to where the regular RNN-T API is located and give some indication of what the expected usage pattern would be, e.g. to call this after doing the regular RNN-T recursion.

I would have thought that alignment code would require a loop of some kind(?), over time, so I'm surprised to not see that here, at least from glancing at it.